server/src/main/java/org/elasticsearch/plugins/PluginsLoader.java

```diff
@@ -118,14 +118,30 @@ public static LayerAndLoader ofLoader(ClassLoader loader) {
      * @param modulesDirectory The directory modules exist in, or null if modules should not be loaded from the filesystem
      * @param pluginsDirectory The directory plugins exist in, or null if plugins should not be loaded from the filesystem
      */
-    @SuppressWarnings("this-escape")
-    public PluginsLoader(Path modulesDirectory, Path pluginsDirectory) {
-        Map<String, List<ModuleQualifiedExportsService>> qualifiedExports = new HashMap<>(ModuleQualifiedExportsService.getBootServices());
-        addServerExportsService(qualifiedExports);
+    public static PluginsLoader createPluginsLoader(Path modulesDirectory, Path pluginsDirectory) {
+        return createPluginsLoader(modulesDirectory, pluginsDirectory, true);
+    }
+
+    /**
+     * Constructs a new PluginsLoader
+     *
+     * @param modulesDirectory The directory modules exist in, or null if modules should not be loaded from the filesystem
+     * @param pluginsDirectory The directory plugins exist in, or null if plugins should not be loaded from the filesystem
+     * @param withServerExports {@code true} to add server module exports
+     */
+    public static PluginsLoader createPluginsLoader(Path modulesDirectory, Path pluginsDirectory, boolean withServerExports) {
```

Comment:
Nit: not a fan of booleans :) 
I'd prefer using explicit method names, like `PluginsLoader createPluginsLoaderWithServerExports` and pass down the map to a common private (or package-private) method, 

```
public static PluginsLoader createPluginsLoaderWithNoAdditionalExports(Path modulesDirectory, Path pluginsDirectory) {     
     return createPluginsLoader(modulesDirectory, pluginsDirectory, Collections.emptyMap());
}

public static PluginsLoader createPluginsLoaderWithServerExports(Path modulesDirectory, Path pluginsDirectory) {
     var qualifiedExports  = new HashMap<>(ModuleQualifiedExportsService.getBootServices());
     addServerExportsService(qualifiedExports);
     return createPluginsLoader(modulesDirectory, pluginsDirectory, qualifiedExports);
}

// Internal? So we can use it to cover other cases? (Or private)
static PluginsLoader createPluginsLoader(Path modulesDirectory, Path pluginsDirectory, Map<String, List<ModuleQualifiedExportsService>> qualifiedExports) {
    // all current code, lines 140-188
}
```
but this is nitpicking :)

Reply:
I do think this is better than what I have now, but I want to make sure this change gets in. So, perhaps this can be a small follow up PR?

Reply:
Sure!